packages/venia-ui/lib/components/CategorySort/categorySort.js

```diff
@@ -0,0 +1,125 @@
+import React, { useMemo, useState } from 'react';
+import { Check } from 'react-feather';
+import uuidV4 from 'uuid/v4';
+import { arrayOf, shape, string, func, oneOf } from 'prop-types';
+
+import defaultClasses from './categorySort.css';
+import { mergeClasses } from '../../classify';
+import Icon from '../Icon/icon';
+
+const CategorySort = props => {
+    const classes = mergeClasses(defaultClasses);
+    const { availableSortMethods, sortControl } = props;
+    const { currentSort, setSort } = sortControl;
+
+    const [visible, setVisible] = useState(false);
```

Comment:
We have a utility just for this! It's a hook called `useDropdown`(https://github.com/magento/pwa-studio/blob/develop/packages/peregrine/lib/hooks/useDropdown.js), and it will handle closing the dropdown whenever a click happens outside of the menu. I recommend using it to avoid duplicating that code.